src/script/conversation/ClientMismatchHandler.ts

```diff
@@ -176,40 +184,39 @@ export class ClientMismatchHandler {
    * @param clientFn Function to remove clients
    * @param conversationEntity Conversation entity
    * @param payload Initial payload resulting in a 412
-   * @param verifyDeletedUsers Verifies if users still exist on the backend by checking their active clients
-   * @returns Function array
    */
-  removePayload(
+  async removePayload(
     recipients: UserClients,
-    clientFn: Function,
+    clientFn: (userId: string, clientId: string) => any | Promise<any>,
     conversationEntity: Conversation = undefined,
     payload: NewOTRMessage,
-    verifyDeletedUsers: boolean,
-  ): void[] {
-    const result: void[] = [];
-
-    const removeDeletedUser = (userId: string) => {
+  ): Promise<void> {
+    const removeDeletedUser = async (userId: string): Promise<void> => {
       const clientIdsOfUser = Object.keys(payload.recipients[userId]);
       const noRemainingClients = !clientIdsOfUser.length;
 
-      if (noRemainingClients) {
-        if (conversationEntity !== undefined) {
-          if (conversationEntity.isGroup() && verifyDeletedUsers) {
-            const timestamp = this.serverTimeHandler.toServerTimestamp();
-            const event = EventBuilder.buildMemberLeave(conversationEntity, userId, false, timestamp);
-            this.eventRepository.injectEvent(event);
+      if (noRemainingClients && typeof conversationEntity !== 'undefined' && conversationEntity.isGroup()) {
+        try {
+          const users = await this.userRepository.fetchUsersById([userId]);
+          if (users.length === 0) {
+            throw new Error('No users found');
           }
+        } catch (error) {
```

Comment:
Catching here is not a good idea as it might inject member leave messages when the network request fails. Just check if `users.length === 0` and then inject the event.